### ISSUE TITLE
Assure property exists

### ISSUE THREAD

**Issue by OliverSanftleben** [pact-foundation/pact-specification#58]
I'm not very satisfied with the matchers. With current matchers it's possible to say that if a property exists it has to be of a special type or must have a special content, but it's not possible to assure, that the property has to exist at all. For example, how would you assure with matchers that with

```
body: {
  "abc": "def"
}
```

property "abc" is a string and must be present in the body?

matching rule

```
"$.body.*": 
{
"match": "type"
},
```

would not give an error, if only

`{} `

is sent.

**Comment by mefellows** [comment]
> With current matchers it's possible to say that if a property exists it has to be of a special type or must have a special content, but it's not possible to assure, that the property has to exist at all

This is incorrect. The problem is, the matcher you've identified is too broad and is not being specific about what it needs. A matcher such as the following would satisfy your request:

```
"$.body.abc": 
{
"match": "type"
}
```

How did you arrive at the above matching rule?

**Comment by OliverSanftleben** [comment]
Is not working for me, if "abc" is missing, test is also ok with this rule.

There is also a problem with arrays, let's say you have the matcher

```
"$.body.array": 
{
  "min": 1,
  "match": "type"
}
```

It works perfect, as long as 'array' is present in the body. But if property 'array' is missing, test says also everything is ok.

I think there is the general problem, that if no matcher can be selected, the test will pass.

**Comment by mefellows** [comment]
Can you please share the code you have that is resulting in this, or even
better, a project so we can repro? What you're asking for is absolutely
fundamental to how Pact works, so something must be going wrong along the
line somewhere.
